Content.Server/Projectiles/ProjectileSystem.cs

```diff
@@ -22,6 +28,9 @@ public override void Initialize()
     {
         base.Initialize();
         SubscribeLocalEvent<ProjectileComponent, StartCollideEvent>(OnStartCollide);
+        SubscribeLocalEvent<DestructibleComponent, DestructionEventArgs>((embeddee, _, _) => UnEmbedChildren(embeddee));
+        SubscribeLocalEvent<BodyComponent, BeingGibbedEvent>((embeddee, _, _) => UnEmbedChildren(embeddee));
+        SubscribeLocalEvent<FoodComponent, BeforeFullyEatenEvent>(OnBeforeFullyEaten);
```

Comment:
`BeforeFullyEatenEvent` seems to be an attempt-event, so it shouldn't ever have any effect. You can only know if it cancelled after the event was raised and all event handlers had a chance to cancel it.

The event only seems to be raised in `DeleteAndSpawnTrash()`, which raises a destruction event anyways, so you should be able to just remove this subscription

Reply:
Oh wow, actually that destruction event was added since I branched -- nice callout.